### PR TITLE
Tegra changes from downstream v3

### PR DESCRIPTION
**Pull request by vwadekar**
Tegra changes from the downstream branch. Batch #3

### CODE REVIEW

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by danh-arm** [comment]
Could you separate the Trusty (and possibly the TLK) SPD changes into a separate PR? They are somewhat "buried" in this PR and seem to be independent to the platform changes.

File: plat/nvidia/tegra/soc/t132/plat_setup.c [hunk 1]
```diff
@@ -29,7 +29,11 @@
  */
 
 #include <xlat_tables.h>
```
Comment (antonio-nino-diaz-arm) [inline plat/nvidia/tegra/soc/t132/plat_setup.c]: This is repeated and not in the correct order.

**Comment by antonio-nino-diaz-arm** [comment]
There are a few coding style errors and warnings. Please, fix the errors (it would be nice to fix the warnings too).

Coding style:

```
plat/nvidia/tegra/common/tegra_bl31_setup.c:142: WARNING: Comparisons should place the constant on the right side of the test
+	if ((BL31_BASE != plat_bl31_params_from_bl2.tzdram_base) &&

plat/nvidia/tegra/common/tegra_gic.c:151: WARNING: line over 80 characters
+				mmio_write_8(gicd_base + GICD_IPRIORITYR + irq_num,

plat/nvidia/tegra/common/tegra_gic.c:152: WARNING: line over 80 characters
+					GIC_HIGHEST_SEC_PRIORITY & GIC_PRI_MASK);

services/spd/trusty/trusty.c:91: WARNING: Missing a blank line after declarations
+	uint64_t r0;
+	r0 = read_hcr();

services/spd/trusty/trusty.c:110: ERROR: do not use C99 // comments
+		//according ARM DEN0028A, VMID stored in x7.

services/spd/trusty/trusty.c:115: ERROR: do not use C99 // comments
+	//r4, r5, r6 reserved for future use.

services/spd/trusty/trusty.c:242: WARNING: line over 80 characters
+			SMC_RET8(handle, ret.r0, ret.r1, ret.r2, ret.r3, ret.r4, ret.r5, ret.r6, ret.r7);

services/spd/trusty/trusty.c:82: ERROR: do not initialise statics to 0
+static uint32_t current_vmid = 0;
```

Include order:

`ERROR:f992fcb:plat/nvidia/tegra/soc/t132/plat_setup.c:System includes not in order.`

**Comment by davidcunado-arm** [comment]
@vwadekar the checks that failed were the static checks - below is the full list of issues.
As per Dan's comments, if you could split the the PR as suggested - I'll then review, etc. 

  CHECKING STYLE
drivers/io/io_memmap.c:136: WARNING: Duplicate signature
Signed-off-by: Douglas Raillard <douglas.raillard@arm.com>

services/spd/tspd/tspd_main.c:642: WARNING: Duplicate signature
Signed-off-by: Douglas Raillard <douglas.raillard@arm.com>

plat/nvidia/tegra/platform.mk:54: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_bl31_setup.c:142: WARNING: Comparisons should place the constant on the right side of the test
+	if ((BL31_BASE != plat_bl31_params_from_bl2.tzdram_base) &&

plat/nvidia/tegra/common/tegra_bl31_setup.c:153: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/include/tegra_private.h:93: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_sip_calls.c:82: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/aarch64/tegra_helpers.S:45: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_pm.c:345: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

lib/cpus/aarch64/denver.S:69: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_gic.c:151: WARNING: line over 80 characters
+				mmio_write_8(gicd_base + GICD_IPRIORITYR + irq_num,

plat/nvidia/tegra/common/tegra_gic.c:152: WARNING: line over 80 characters
+					GIC_HIGHEST_SEC_PRIORITY & GIC_PRI_MASK);

plat/nvidia/tegra/soc/t210/plat_setup.c:130: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/include/tegra_private.h:90: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/include/tegra_private.h:103: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/aarch64/tegra_helpers.S:482: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

services/spd/trusty/trusty.c:91: WARNING: Missing a blank line after declarations
+	uint64_t r0;
+	r0 = read_hcr();

services/spd/trusty/trusty.c:110: ERROR: do not use C99 // comments
+		//according ARM DEN0028A, VMID stored in x7.

services/spd/trusty/trusty.c:115: ERROR: do not use C99 // comments
+	//r4, r5, r6 reserved for future use.

services/spd/trusty/trusty.c:242: WARNING: line over 80 characters
+			SMC_RET8(handle, ret.r0, ret.r1, ret.r2, ret.r3, ret.r4, ret.r5, ret.r6, ret.r7);

services/spd/trusty/trusty_helpers.S:83: WARNING: Duplicate signature
Signed-off-by: Anthony Zhou <anzhou@nvidia.com>

services/spd/trusty/trusty_helpers.S:83: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

services/spd/trusty/trusty.c:82: ERROR: do not initialise statics to 0
+static uint32_t current_vmid = 0;

services/spd/trusty/trusty.c:278: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_pm.c:219: WARNING: Duplicate signature
Signed-off-by: Wayne Lin <wlin@nvidia.com>

plat/nvidia/tegra/common/tegra_pm.c:219: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

services/spd/trusty/trusty.c:430: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_bl31_setup.c:219: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/common/tegra_bl31_setup.c:111: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

plat/nvidia/tegra/include/tegra_private.h:94: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

services/spd/tlkd/tlkd_main.c:185: WARNING: Duplicate signature
Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

total: 3 errors, 28 warnings, 1281 lines checked

**Comment by vwadekar** [comment]
Pushed updated branch with the fixes. Please review.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Pull request merged**